### ISSUE TITLE
[Question] How to make some changes to a generated reconclier

### ISSUE THREAD

**Issue by djdongjin** [kubernetes-sigs/kubebuilder-declarative-pattern#235]
Hi, I have a question about _reconcilers built based on kubebuild-declarative-pattern_. Suppose I want that in the reconcile loop, the reconciler should read an annotation from a namespace and based on that annotation value, add a node-affinity (or any other deployment change) to the deployment it creates. Is it possible? Thanks! For example:

1. Reconclier: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/examples/guestbook-operator/controllers/guestbook_controller.go#L55
2. Deployment: https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/examples/guestbook-operator/channels/packages/guestbook/0.1.0/manifest.yaml#L19

Basically, the question is how to make some customization to the generated reconclier.

**Comment by atoato88** [comment]
I think we can do that operation by multiple options.

- option 1: patch manifest by kustomize
guestbook-operator is set with `WithApplyKustomize` at [here](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/2072cf081b2e512b687c6bf152466646df1ac2c9/examples/guestbook-operator/controllers/guestbook_controller.go#L62).
So you can patch target manifest by the way kustomize can.
I created sample codebase which add kustomize patch, [please refer this](https://github.com/atoato88/kubebuilder-declarative-pattern/commit/69f23df8e376354891e1d179489950154e3b4655).
I think it's easier way than next one.

- option 2: implement a func for used by [`WithObjectTransform`](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/master/reconciler-options.md#withobjecttransform)
We can implement any functions for manipulating a target manifest in Go code.
It's more flexible way than previous one, because this way can implement any logics as you like.
please refer codebase in guestbook-operator. [1](), [2]() 
But it's more complicated, I think.


I'm not sure that which one is better for your requirement, but maybe option 2 is better because your requirement seems to have some logics depending on complicated condtion.

**Comment by djdongjin** [comment]
@atoato88 thanks for the suggestion! I finally figured it out using the `WithObjectTransform` and `MutatePodSpec` [here](https://github.com/kubernetes-sigs/kubebuilder-declarative-pattern/blob/2072cf081b2e512b687c6bf152466646df1ac2c9/pkg/patterns/declarative/pkg/manifest/objects.go#L189)

**Comment by atoato88** [comment]
You're welcome, and thank you for sharing your solution. 😉
